src/sagemaker/cli/framework_upgrade.py

```diff
@@ -0,0 +1,248 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""A Python script to upgrade framework versions and regions"""
+from __future__ import absolute_import
+
+import argparse
+import json
+import os
+
+from sagemaker.image_uris import config_for_framework
+
+IMAGE_URI_CONFIG_DIR = os.path.join("..", "image_uri_config")
+
+
+def _get_latest_values(existing_content, scope=None):
+    """Get the latest "registries", "py_versions" and "repository" values
+
+    Args:
+        existing_content (dict): Dictionary of complete framework image information.
+        scope (str): Type of the image, required if the target is DLC
+            framework (Default: None).
+    """
+    if scope in existing_content:
+        existing_content = existing_content[scope]
+    else:
+        if "versions" not in existing_content:
+            raise ValueError(
+                "Invalid image scope: {}. Valid options: {}.".format(
+                    scope, ", ".join(existing_content.key())
+                )
+            )
+
+    latest_version = list(existing_content[scope]["versions"].keys())[-1]
+    registries = existing_content[scope]["versions"][latest_version]["registries"]
+    py_versions = existing_content[scope]["versions"][latest_version]["py_versions"]
+    repository = existing_content[scope]["versions"][latest_version]["repository"]
```

Comment:
I think there is something wrong with this if you've already gotten `existing_content` from `scope` above.

Reply:
You are right, the last 3 lines should not reference "scope" anymore. I've updated the method and created unit tests for it.